### PR TITLE
MAYA-121534 Import code for place2dTexture

### PR DESCRIPTION
**Pull request by JGamache-autodesk**
- Includes import code for fileTexture node as well

### CODE REVIEW

File: lib/usd/translators/shading/mtlxBaseReader.cpp [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+
+PXR_NAMESPACE_OPEN_SCOPE
+
+REGISTER_SHADING_MODE_IMPORT_MATERIAL_CONVERSION(
+    TrMtlxTokens->conversionName,
+    TrMtlxTokens->contextName,
+    TrMtlxTokens->niceName,
+    TrMtlxTokens->importDescription);
+
+TF_REGISTRY_FUNCTION(UsdMayaShaderReaderRegistry)
```
Comment (JGamache-autodesk) [inline lib/usd/translators/shading/mtlxBaseReader.cpp:51]: Moved from mtlxBaseWriter.cpp

File: lib/usd/translators/shading/mtlxBaseReader.cpp [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+    UsdMayaSymmetricShaderReader::RegisterReader(
+        TrMtlxTokens->MayaND_phong_surfaceshader,
+        TrMayaTokens->phong,
+        TrMtlxTokens->conversionName);
+    UsdMayaSymmetricShaderReader::RegisterReader(
+        TrMtlxTokens->MayaND_blinn_surfaceshader,
+        TrMayaTokens->blinn,
+        TrMtlxTokens->conversionName);
+    UsdMayaSymmetricShaderReader::RegisterReader(
+        TrMtlxTokens->MayaND_place2dTexture_vector2,
```
Comment (JGamache-autodesk) [inline lib/usd/translators/shading/mtlxBaseReader.cpp:66]: New one. Symmetric so only requires registration.

File: lib/usd/translators/shading/mtlxBaseReader.cpp [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+
+    return true;
+}
+
+bool MtlxUsd_BaseReader::GetColorAndAlphaFromInput(
+    const UsdShadeShader& shader,
+    const TfToken&        inputName,
+    GfVec3f&              color,
+    float&                alpha)
+{
```
Comment (JGamache-autodesk) [inline lib/usd/translators/shading/mtlxBaseReader.cpp:115]: Extracted from MtlxUsd_ImageReader::Read( ).

File: lib/usd/translators/shading/mtlxImageReader.cpp [hunk 4]
```diff
@@ -88,25 +91,31 @@ bool MtlxUsd_ImageReader::Read(UsdMayaPrimReaderContext& context)
...
+    ;
+    MStatus status;
+    if (mayaObject.isNull()) {
+        if (!UsdMayaTranslatorUtil::CreateShaderNode(
+                MString(prim.GetName().GetText()),
+                TrMayaTokens->file.GetText(),
+                UsdMayaShadingNodeType::Texture,
+                &status,
+                &mayaObject)) {
+            // we need to make sure assumes those types are loaded..
```
Comment (boudrey) [inline lib/usd/translators/shading/mtlxImageReader.cpp:105]: make sure or assume?

File: lib/usd/translators/shading/mtlxImageReader.cpp [hunk 5]
```diff
@@ -88,25 +91,31 @@ bool MtlxUsd_ImageReader::Read(UsdMayaPrimReaderContext& context)
...
-          && depFn.setObject(mayaObject))) {
-        // we need to make sure assumes those types are loaded..
-        TF_RUNTIME_ERROR(
-            "Could not create node of type '%s' for shader '%s'.\n",
-            TrMayaTokens->file.GetText(),
-            prim.GetPath().GetText());
-        return false;
+    // It is possible the file node already exists if we encountered a post-processor:
+    MObject mayaObject = context.GetMayaNode(prim.GetPath(), false);
+    ;
```
Comment (boudrey) [inline lib/usd/translators/shading/mtlxImageReader.cpp:96]: Empty line?

**Review by boudrey — APPROVED** [review]

**Pull request merged**